### PR TITLE
Assorted python updates related to python310

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Fix two packages for python3.10 that were not currently building.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rmcgibbo** [comment]
[bump]

**Comment by wd15** [comment]
`python310Package.mpi4py` doesn't build on master, but does now build on this branch.

A lot of other packages seems to be broken though.

Result of `nixpkgs-review pr 154197` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)                                                                                          
<details>                                                                                                                                                                                      
  <summary>26 packages marked as broken and skipped:</summary>                                                                                                                                 
  <ul>
    <li>python310Packages.dask-xgboost</li>
    <li>python310Packages.dm-haiku</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.graph_nets</li>
    <li>python310Packages.hickle</li>
    <li>python310Packages.nipype</li>
    <li>python310Packages.optax</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.pybids</li>
    <li>python310Packages.pywick</li>
    <li>python310Packages.rl-coach</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.treex</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.dm-sonnet</li>
    <li>python39Packages.graph_nets</li>
    <li>python39Packages.hickle</li>
    <li>python39Packages.nipype</li>
    <li>python39Packages.optuna</li>
    <li>python39Packages.pybids</li>
    <li>python39Packages.pymc3</li>
    <li>python39Packages.pywick</li>
    <li>python39Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>154 packages failed to build:</summary>
  <ul>
    <li>diffoscope</li>
    <li>gpt2tc</li>
    <li>intensity-normalization (python39Packages.intensity-normalization)</li>
    <li>napari (python39Packages.napari)</li>
    <li>paraview</li>
    <li>python310Packages.annoy</li>
    <li>python310Packages.aplpy</li>
    <li>python310Packages.asdf</li>
    <li>python310Packages.batchgenerators</li>
    <li>python310Packages.bayespy</li>
    <li>python310Packages.caffe</li>
    <li>python310Packages.catalogue</li>
    <li>python310Packages.dask-gateway</li>
    <li>python310Packages.dask-image</li>
    <li>python310Packages.dask-jobqueue</li>
    <li>python310Packages.dask-mpi</li>
    <li>python310Packages.dicom2nifti</li>
    <li>python310Packages.dipy</li>
    <li>python310Packages.distributed</li>
    <li>python310Packages.env-canada</li>
    <li>python310Packages.fenics</li>
    <li>python310Packages.fipy</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5py</li>
    <li>python310Packages.h5py-mpi</li>
    <li>python310Packages.handout</li>
    <li>python310Packages.hoomd-blue</li>
    <li>python310Packages.image-match</li>
    <li>python310Packages.imagecodecs-lite</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imageio</li>
    <li>python310Packages.imgaug</li>
    <li>python310Packages.intensity-normalization</li>
    <li>python310Packages.keras</li>
    <li>python310Packages.keras-applications</li>
    <li>python310Packages.meshio</li>
    <li>python310Packages.moviepy</li>
    <li>python310Packages.mpi4py</li>
    <li>python310Packages.napari</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.napari-svg</li>
    <li>python310Packages.nibabel</li>
    <li>python310Packages.nilearn</li>
    <li>python310Packages.nipy</li>
    <li>python310Packages.nitime</li>
    <li>python310Packages.nitransforms</li>
    <li>python310Packages.phonopy</li>
    <li>python310Packages.pims</li>
    <li>python310Packages.pyregion</li>
    <li>python310Packages.pytest-astropy</li>
    <li>python310Packages.pytest-astropy-header</li>
    <li>python310Packages.python-openems</li>
    <li>python310Packages.qiskit-aqua</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.reproject</li>
    <li>python310Packages.runway-python</li>
    <li>python310Packages.scikitimage</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.slicedimage</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-pkuseg</li>
    <li>python310Packages.spectral-cube</li>
    <li>python310Packages.srsly</li>
    <li>python310Packages.streamz</li>
    <li>python310Packages.sumo</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.thinc</li>
    <li>python310Packages.tifffile</li>
    <li>python310Packages.worldengine</li>
    <li>python39Packages.annoy</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.arviz</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.bayespy</li>
    <li>python39Packages.caffe</li>
    <li>python39Packages.catalogue</li>
    <li>python39Packages.clifford</li>
    <li>python39Packages.cppe</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-image</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-ml</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.datashader</li>
    <li>python39Packages.dicom2nifti</li>
    <li>python39Packages.dipy</li>
    <li>python39Packages.distributed</li>
    <li>python39Packages.dm-haiku</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.env-canada</li>
    <li>python39Packages.fenics</li>
    <li>python39Packages.fipy</li>
    <li>python39Packages.flax</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.h5netcdf</li>
    <li>python39Packages.h5py</li>
    <li>python39Packages.h5py-mpi</li>
    <li>python39Packages.handout</li>
    <li>python39Packages.hoomd-blue</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecodecs-lite</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.imageio</li>
    <li>python39Packages.imgaug</li>
    <li>python39Packages.keras</li>
    <li>python39Packages.keras-applications</li>
    <li>python39Packages.meshio</li>
    <li>python39Packages.moviepy</li>
    <li>python39Packages.mpi4py</li>
    <li>python39Packages.napari-console</li>
    <li>python39Packages.napari-svg</li>
    <li>python39Packages.nibabel</li>
    <li>python39Packages.nilearn</li>
    <li>python39Packages.nipy</li>
    <li>python39Packages.nitime</li>
    <li>python39Packages.nitransforms</li>
    <li>python39Packages.optax</li>
    <li>python39Packages.phonopy</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pyscf</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.python-openems</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aqua</li>
    <li>python39Packages.qiskit-nature</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.runway-python</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.sfepy</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-pkuseg</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.srsly</li>
    <li>python39Packages.streamz</li>
    <li>python39Packages.stumpy</li>
    <li>python39Packages.stytra</li>
    <li>python39Packages.sumo</li>
    <li>python39Packages.sunpy</li>
    <li>python39Packages.test-tube</li>
    <li>python39Packages.textacy</li>
    <li>python39Packages.thinc</li>
    <li>python39Packages.tifffile</li>
    <li>python39Packages.treex</li>
    <li>worldengine-cli (python39Packages.worldengine)</li>
    <li>tartube</li>
    <li>tartube-yt-dlp</li>
    <li>veusz</li>
    <li>visidata</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python310Packages.glymur</li>
  </ul>
</details>

**Comment by rmcgibbo** [comment]
@wd15: Can you share any build failure logs? All of the updated packages build here on this PR for me (` nix build -f . python310Packages.catalogue python310Packages.h5py python310Packages.imagecodecs-lite python310Packages.mpi4py python310Packages.sfepy  --keep-going`), but it looks like they didn't for you.

**Comment by wd15** [comment]
All packages listed seem to build now after the latest changes (using nix-build). I don't know about dependencies as nixpkgs-review seems to take forever. I'll try running it overnight.

**Comment by wd15** [comment]
After running nixpkgs-review the only packages that fail to build are the following

   * python310Packages.caffe
   * python310Packages.fenics
   * python310Packages.fipy
   * python310Packages.nilearn
   * python310Packages.qiskit-aqua
   * python310Packages.qiskit-nature
   * python310Packages.spacy
   * python310Packages.textacy
   * python310Packages.thinc
   * python39Packages.fenics
   * python39Packages.qiskit 
   * python39Packages.spacy
   * python39Packages.spacy-transformers
   * python39Packages.textacy
   * python39Packages.thinc

All of those packages fail to build on master so no regressions in this PR as far as I can tell.

**Review by wd15 — APPROVED** [review]
